### PR TITLE
[Heartbeat] Fix origin, Cleanup stdfields

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes https://github.com/elastic/beats/issues/31870 and refactors the wrapper code to be cleaner and more consistent.

Prior to this PR we would only set the `monitor.origin` field if the `monitor.origin` config setting was set on non-browser (e.g. HTTP) monitors. The reason the original patch only worked on lightweight monitors was that there has historically been a separate codepath for adding simple core metadata fields to lightweight monitors than browser based monitors.

This PR removes that codepath, funneling the writing of most browser `monitor.*` fields through `addMonitorMeta` in `wrappers.go`. This means that we'll have more consistent behavior between browser and regular monitors going forward.

This entailed some untangling of some funky code to get working, apologies for the larger diff, but it leaves the codebase more consistent than we found it.

One weird wrinkle is the `monitor.project` field, which we don't currently use, but customers might. This contains the literal `id` and `name` from the monitors YAML when running `zip_url` or `local` monitors. This does keep the 8.3 change moving this from `suite.{id,name}` to `monitor.project.{id,name}` but does it in a cleaner way, labeling as many relevant variables, tests, and functions as legacy. One day we can remove this when / if we remove support for `zip_url` and `local` browser sources.

We should probably do this before the beta is over.

This PR also tries to more consistently refer to instances of `stdfields.StandardFields` as `sFields` when used as a var.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Try running monitors with `origin` set, as well as some `zip_url` and `local` monitors.

### CODE REVIEW

**Comment by shahzad31** [comment]
It sets the 'monitor.origin' field now

image

**Review by shahzad31 — APPROVED** [review]
Tested and it looks good !!

Code changes also makes sense !!

I didn't e2e tested changes related to legacy zip_url source

File: heartbeat/monitors/wrappers/wrappers_test.go [hunk 1]
```diff
@@ -526,17 +530,26 @@ var projectMonitorValues = BrowserMonitor{
 	checkGroup: "journey-1-check-group",
 }
 
-func makeProjectBrowserJob(t *testing.T, u string, summary bool, projectErr error) jobs.Job {
+// Used for testing legacy zip_url / local monitorss
```
Comment (lucasfcosta) [inline heartbeat/monitors/wrappers/wrappers_test.go:533]: Nit: 
```suggestion
// Used for testing legacy zip_url / local monitors
```

**Review by lucasfcosta — COMMENTED** [review]
I know this has already been merged, but thought I'd do another pass anyway just so I'd be up-to-date on these changes.

Really glad you moved the appending of project fields and origin to the wrapper above. I found that part of the code to be extremely useful when I was writing the logging/telemetry piece too.

It took some time to wrap my head about how the wrappers get applied (pun intended) when I was first reading the code, but once I understood I thought it was great to centralize that.

**Comment by andrewvc** [comment]
@Mergifyio backport 8.3

**Pull request merged**